shepherd_bms/src/main.cpp

```diff
@@ -18,94 +19,86 @@ bool dischargeConfig[NUM_CHIPS][NUM_CELLS_PER_CHIP] = {};
 ChipData_t *testData;
 Timer mainTimer;
 ComputeInterface compute;
+WDT_T4<WDT1> wdt;
+
+Timer chargeTimeout;
+Timer boostTimer;
+Timer boostRechargeTimer;
+
+AccumulatorData_t *prevAccData = nullptr;
 
 uint32_t bmsFault = FAULTS_CLEAR;
 
-void testSegments()
+uint16_t overVoltCount = 0;
+uint16_t underVoltCount = 0;
+uint16_t overCurrCount = 0;
+uint16_t chargeOverVolt = 0;
+uint16_t overChgCurrCount = 0;
+
+enum
 {
-	currTime = millis();
-	if (lastVoltTemp + 1000 < currTime) {
-		lastVoltTemp = currTime;
-		//Handle Segment data collection test logic
-		testData = new ChipData_t[NUM_CHIPS];
+	BOOST_STANDBY,
+	BOOSTING,
+	BOOST_RECHARGE
+}BoostState;
 
-		//Retrieve ALL segment data (therms and voltage)
-		segment.retrieveSegmentData(testData);
 
-		for (int chip = 0; chip < NUM_CHIPS; chip++)
-		{
-			for (int cell=0; cell < NUM_CELLS_PER_CHIP; cell++)
-			{
-				Serial.print(testData[chip].voltageReading[cell]);
-				Serial.print("\t");
-			}
-
-			Serial.println(); //newline
+void chargeBalancing(AccumulatorData_t *bms_data)
+{
+	bool balanceConfig[NUM_CHIPS][NUM_CELLS_PER_CHIP];
 
-			for (int therm=17; therm < 28; therm++)
-			{
-				Serial.print(testData[chip].thermistorReading[therm]);
-				Serial.print("\t");
-				if (therm == 15) Serial.println();
-			}
-			Serial.println(); //newline
+    for(uint8_t chip = 0; chip < NUM_CHIPS; chip++)
+    {
+		for(uint8_t cell = 0; cell < NUM_CELLS_PER_CHIP; cell++)
+		{
+			uint16_t delta = bms_data->chipData[chip].voltageReading[cell] - (uint16_t)bms_data-> minVoltage.val;
+			if(delta > MAX_DELTA_V * 10000)
+				balanceConfig[chip][cell] = true;
+			else
+				balanceConfig[chip][cell] = false;
+        }
+    }
+	
+	/*Serial.println("Cell Balancing:");
```

Comment:
is this eventually going to be used? commented out code in main is not ideal, if able to we should delete if not at least add comment to explain what's going on

Reply:
Should be moved into whatever DEBUG print functions we want